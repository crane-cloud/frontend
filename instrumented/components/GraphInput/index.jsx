import React, { useState, useEffect, useCallback } from "react";
import PropTypes from "prop-types";
import Calendar from "../Calendar";
import PrimaryButton from "../PrimaryButton";
import {
  monthNames,
  today,
  currentMonth,
  currentYear,
} from "../../helpers/dateConstants";
import "./GraphInput.css";

const GraphInput = ({
  handleChange,
  position,
  label,
  showCalendar,
  onClick,
  value,
  onCancel,
  onSubmit,
}) => {
  const [date, setDate] = useState({
    day: today,
    month: currentMonth,
    year: currentYear,
  });

  const formatString = (num) => {
    // appends a leading 0 in case of 1-length string
    const numString = num.toString();

    if (numString.length < 2) {
      return `0${numString}`;
    }

    return numString;
  };

  const createTimestamp = useCallback(() => {
    const { year, month, day } = date;
    const dateString = `${year}-${formatString(month + 1)}-${formatString(
      day
    )}`;
    const timeStamp = Date.parse(`${dateString}`);

    if (showCalendar) {
      // only call active parent
      handleChange(timeStamp);
    }
  }, [date, showCalendar, handleChange]);

  const getDate = ({ day, month, year }) => {
    setDate({
      ...date,
      day,
      month,
      year,
    });
  };

  const trimMonthName = (month) => month.substring(0, 3);

  useEffect(() => {
    createTimestamp();
  }, [createTimestamp]);

  return (
    <div className="DateInputContainer">
      <div className="DateInputWrapper">
        <div className="DateInputLabel">{label}</div>
        <div
          className={`DateInputDisplay ${showCalendar && "DisplayActive"}`}
          value={value}
          onClick={onClick}
          role="presentation"
        >
          {date
            ? `${trimMonthName(monthNames[date.month])} ${date.day}, ${
                date.year
              }`
            : `${trimMonthName(
                monthNames[currentMonth]
              )} ${today}, ${currentYear}`}
        </div>
      </div>
      {showCalendar && (
        <div className={`DateInputCalendar ${position}`}>
          <Calendar onChange={getDate} />
          <div className="CalendarModalButtons">
            <PrimaryButton className="CancelBtn ModalBtn" onClick={onCancel}>
              Cancel
            </PrimaryButton>
            <PrimaryButton className="ModalBtn" onClick={onSubmit}>
              Proceed
            </PrimaryButton>
          </div>
        </div>
      )}
    </div>
  );
};

GraphInput.defaultProps = {
  label: "",
  position: "",
};

GraphInput.propTypes = {
  label: PropTypes.string,
  position: PropTypes.string,
  handleChange: PropTypes.func.isRequired,
  showCalendar: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};

export default GraphInput;
