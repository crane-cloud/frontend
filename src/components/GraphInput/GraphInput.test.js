/* eslint-disable no-undef */
import React from "react";
import { shallow } from "enzyme";
import GraphInput from ".";
import { bool } from "prop-types";
const GraphInputProps = {
    formatString: jest.fn(),
    createTimestamp: jest.fn(),
    handleChange: jest.fn(),
    onClick: jest.fn(),
    showCalendar: bool,
    getDate: jest.fn(),
    trimMonthName: jest.fn(),
};
describe('Graph input component', () => {
  const GraphInputWrapper = shallow(<GraphInput {...GraphInputProps} />);

  it("graph input  should match snapshot", () => {
    expect(GraphInputWrapper).toMatchSnapshot();
  });
  it("calls the onClick callback when the input is clicked", () => {
    const onClickMock = jest.fn();
    const wrapper = shallow(
      <GraphInput
        handleChange={jest.fn()}
        showCalendar={false}
        onClick={onClickMock}
        value=""
      />
    );
  
    wrapper.find(".DateInputDisplay").simulate("click");
    expect(onClickMock).toHaveBeenCalled();
  });
  // it("updates the state correctly when the Calendar changes", () => {
  //   const wrapper = shallow(
  //     <GraphInput
  //       handleChange={jest.fn()}
  //       showCalendar={true}
  //       onClick={jest.fn()}
  //       value=""
  //     />
  //   );

  //   // Trigger the change event on the Calendar component
  //   wrapper.find("Calendar").prop("onChange")({ day: 1, month: 2, year: 2023 });

  //   // Get the state value using the useState hook
  //   const [date] = wrapper.find("GraphInput").prop("date");

  //   // Verify that the state of the GraphInput component is updated correctly
  //   expect(date).toEqual({ day: 1, month: 2, year: 2023 });
  // });

  it("calls the onCancel callback when the cancel button is clicked", () => {
    const onCancelMock = jest.fn();
    const wrapper = shallow(
      <GraphInput
        handleChange={jest.fn()}
        showCalendar={true}
        onClick={jest.fn()}
        value=""
        onCancel={onCancelMock}
      />
    );
  
    wrapper.find(".CancelBtn").simulate("click");
    expect(onCancelMock).toHaveBeenCalled();
  });

  it("should match the snapshot", () => {
    GraphInputWrapper.setProps(GraphInputProps);
    expect(GraphInputWrapper).toBeDefined();
    expect(GraphInputWrapper).toHaveLength(1);
  });
});
