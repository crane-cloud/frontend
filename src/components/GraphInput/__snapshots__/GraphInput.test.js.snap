// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graph input component graph input  should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="DateInputContainer"
      >
        <div
          class="DateInputWrapper"
        >
          <div
            class="DateInputLabel"
          />
          <div
            class="DateInputDisplay DisplayActive"
            role="presentation"
          >
            Dec 21, 2023
          </div>
        </div>
        <div
          class="DateInputCalendar "
        >
          <div
            class="CalendarWrapper DisableTextSelect"
          >
            <div
              class="CalendarMonth"
            >
              <div
                class="CalendarArrow"
                role="presentation"
              >
                <iconmock />
              </div>
              December 2023
              <div
                class="CalendarArrow"
                role="presentation"
              >
                <iconmock />
              </div>
            </div>
            <div
              class="CalendarDayNames"
            >
              <div
                class="WeekDay"
              >
                sun
              </div>
              <div
                class="WeekDay"
              >
                mon
              </div>
              <div
                class="WeekDay"
              >
                tue
              </div>
              <div
                class="WeekDay"
              >
                wed
              </div>
              <div
                class="WeekDay"
              >
                thu
              </div>
              <div
                class="WeekDay"
              >
                fri
              </div>
              <div
                class="WeekDay"
              >
                sat
              </div>
            </div>
            <div
              class="CalendarWeeks"
            >
              <div
                class="CalendarWeekDays"
              >
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div />
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div />
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div />
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div />
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div />
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    1
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    2
                  </div>
                </div>
              </div>
              <div
                class="CalendarWeekDays"
              >
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    3
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    4
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    5
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    6
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    7
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    8
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    9
                  </div>
                </div>
              </div>
              <div
                class="CalendarWeekDays"
              >
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    10
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    11
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    12
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    13
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    14
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    15
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    16
                  </div>
                </div>
              </div>
              <div
                class="CalendarWeekDays"
              >
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    17
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    18
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    19
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    20
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  Today
                  "
                  role="presentation"
                >
                  <div>
                    21
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    22
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    23
                  </div>
                </div>
              </div>
              <div
                class="CalendarWeekDays"
              >
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    24
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    25
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    26
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    27
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    28
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    29
                  </div>
                </div>
                <div
                  class="
                  CalendarSingleDay 
                  false
                  "
                  role="presentation"
                >
                  <div>
                    30
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="CalendarModalButtons"
          >
            <button
              class="Primary-Btn CancelBtn ModalBtn false
        undefined undefined undefined
        undefined
      "
            >
              Cancel
            </button>
            <button
              class="Primary-Btn ModalBtn false
        undefined undefined undefined
        undefined
      "
            >
              Proceed
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="DateInputContainer"
    >
      <div
        class="DateInputWrapper"
      >
        <div
          class="DateInputLabel"
        />
        <div
          class="DateInputDisplay DisplayActive"
          role="presentation"
        >
          Dec 21, 2023
        </div>
      </div>
      <div
        class="DateInputCalendar "
      >
        <div
          class="CalendarWrapper DisableTextSelect"
        >
          <div
            class="CalendarMonth"
          >
            <div
              class="CalendarArrow"
              role="presentation"
            >
              <iconmock />
            </div>
            December 2023
            <div
              class="CalendarArrow"
              role="presentation"
            >
              <iconmock />
            </div>
          </div>
          <div
            class="CalendarDayNames"
          >
            <div
              class="WeekDay"
            >
              sun
            </div>
            <div
              class="WeekDay"
            >
              mon
            </div>
            <div
              class="WeekDay"
            >
              tue
            </div>
            <div
              class="WeekDay"
            >
              wed
            </div>
            <div
              class="WeekDay"
            >
              thu
            </div>
            <div
              class="WeekDay"
            >
              fri
            </div>
            <div
              class="WeekDay"
            >
              sat
            </div>
          </div>
          <div
            class="CalendarWeeks"
          >
            <div
              class="CalendarWeekDays"
            >
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div />
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div />
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div />
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div />
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div />
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  1
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  2
                </div>
              </div>
            </div>
            <div
              class="CalendarWeekDays"
            >
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  3
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  4
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  5
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  6
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  7
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  8
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  9
                </div>
              </div>
            </div>
            <div
              class="CalendarWeekDays"
            >
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  10
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  11
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  12
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  13
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  14
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  15
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  16
                </div>
              </div>
            </div>
            <div
              class="CalendarWeekDays"
            >
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  17
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  18
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  19
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  20
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  Today
                  "
                role="presentation"
              >
                <div>
                  21
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  22
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  23
                </div>
              </div>
            </div>
            <div
              class="CalendarWeekDays"
            >
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  24
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  25
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  26
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  27
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  28
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  29
                </div>
              </div>
              <div
                class="
                  CalendarSingleDay 
                  false
                  "
                role="presentation"
              >
                <div>
                  30
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="CalendarModalButtons"
        >
          <button
            class="Primary-Btn CancelBtn ModalBtn false
        undefined undefined undefined
        undefined
      "
          >
            Cancel
          </button>
          <button
            class="Primary-Btn ModalBtn false
        undefined undefined undefined
        undefined
      "
          >
            Proceed
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
